src/import-export/csv-imp/gnc-tx-import.cpp

```diff
@@ -0,0 +1,960 @@
+                                 "(?<DAY>[0-9]+)"
+                             "|"                                     // or CCYYMMDD
+                                 "(?<YEAR>[0-9]{4})"
+                                 "(?<MONTH>[0-9]{2})"
+                                 "(?<DAY>[0-9]{2})"
+                             ")",
+
+                             "(?:"                                   // either d-m-y
+                                 "(?<DAY>[0-9]+)[-/.' ]+"
+                                 "(?<MONTH>[0-9]+)[-/.' ]+"
+                                 "(?<YEAR>[0-9]+)"
+                             "|"                                     // or DDMMCCYY
+                                 "(?<DAY>[0-9]{2})"
+                                 "(?<MONTH>[0-9]{2})"
+                                 "(?<YEAR>[0-9]{4})"
+                             ")",
+
+                             "(?:"                                   // either m-d-y
+                                 "(?<MONTH>[0-9]+)[-/.' ]+"
+                                 "(?<DAY>[0-9]+)[-/.' ]+"
+                                 "(?<YEAR>[0-9]+)"
+                             "|"                                     // or MMDDCCYY
+                                 "(?<MONTH>[0-9]{2})"
+                                 "(?<DAY>[0-9]{2})"
+                                 "(?<YEAR>[0-9]{4})"
+                             ")",
+
+                             "(?:"                                   // either d-m(-y)
+                                 "(?<DAY>[0-9]+)[-/.' ]+"
+                                 "(?<MONTH>[0-9]+)(?:[-/.' ]+"
+                                 "(?<YEAR>[0-9]+))?"
+                             "|"                                     // or DDMM(CCYY)
+                                 "(?<DAY>[0-9]{2})"
+                                 "(?<MONTH>[0-9]{2})"
+                                 "(?<YEAR>[0-9]+)?"
+                             ")",
+
+                             "(?:"                                   // either m-d(-y)
+                                 "(?<MONTH>[0-9]+)[-/.' ]+"
+                                 "(?<DAY>[0-9]+)(?:[-/.' ]+"
+                                 "(?<YEAR>[0-9]+))?"
+                             "|"                                     // or MMDD(CCYY)
+                                 "(?<MONTH>[0-9]{2})"
+                                 "(?<DAY>[0-9]{2})"
+                                 "(?<YEAR>[0-9]+)?"
+                             ")",
+};
+//const int num_currency_formats = 3;
+//const gchar* currency_format_user[] = {N_("Locale"),
+//                                       N_("Period: 123,456.78"),
+//                                       N_("Comma: 123.456,78")
+//                                      };
+//
+/* This map contains a set of strings representing the different column types. */
+std::map<GncTransPropType, const char*> gnc_csv_col_type_strs = {
```

Comment:
I used an enum class because Meyers explains it's preferred for stronger type-safety. The only use for an old-style enum he proposes is to access members of a tuple.
In case of an unscoped enum, it's not even necessary to specify the underlying type (again according to Meyers). It should implicitly convert anyway.
I'll defer this conversion until it turns out to be a performance issue. On my machine it's still blazingly fast right now (but I haven't done any profiling yet).